src/providers/spatialite/qgsspatialiteprovider.cpp

```diff
@@ -4380,6 +4406,19 @@ QVariant QgsSpatiaLiteProvider::defaultValue( int fieldId ) const
   return mDefaultValues.value( fieldId, QVariant() );
 }
 
+bool QgsSpatiaLiteProvider::skipConstraintCheck( int fieldIndex, QgsFieldConstraints::Constraint constraint, const QVariant &value ) const
+{
+  Q_UNUSED( constraint );
+
+  // If the field is the primary key, skip in case it's autog-enerated / auto-incrementing
+  if ( mAttributeFields.at( fieldIndex ).name() == mPrimaryKey  && mPrimaryKeyAutoIncrement )
+  {
+    return mDefaultValues.value( fieldIndex, QVariant() ) != value;
```

Comment:
when comparing QVariants it's often required to check NULL values separately.

    const QVariant defVal = mDefaultValues.value( fieldIndex );
    return defVal != value && defVal.isNull() != value.isNull();

Because for Qt `QVariant( 0 ) == QVariant( QVariant::Integer )`